docs/en/writing-running-appium/ios/ios-xctest-file-movement.md

```diff
@@ -0,0 +1,130 @@
+## Pushing/Pulling files
+
+Appium provides [Pull Folder](http://appium.io/docs/en/commands/device/files/pull-folder/), [Pull File](http://appium.io/docs/en/commands/device/files/pull-file/) and [Push File](http://appium.io/docs/en/commands/device/files/push-file/) to move files.
+This documentation aims to help to understand how they work for iOS.
+
+### Format
+
+Below is the basic format.
+
+1. `@<app_bundle_id>:<optional_container_type>/<path_to_the_file_or_folder_inside_container>`
+2. `@<app_bundle_id>/<path_to_the_file_or_folder_inside_container>`
+3. `<path_to_the_file_or_folder_inside_container>`
+
+### Real device
+
+For real devices, Appium relies on [ifuse](https://github.com/libimobiledevice/ifuse) to achieve the feature.
+You may compile ifuse from the source code or via brew.
+
+```
+$ brew cask install osxfuse
+$ brew install ifuse --HEAD # newer iOS versions need the latest codebase
+```
+
+#### Format
+
+The format of method argument should be the following:
+
+- `@<app_bundle_id>` is a bundle id
+- `optional_container_type` is container type
+    - `documents` is only available
+        - BundleIds which can get by `ifuse -u <udid> --list-apps` can specify `documents` type
+        - e.g. Below _On My iPhone_ image has _Slack_ folder, but `com.tinyspeck.chatlyio` does not exist in the output of `--list-apps`. Thus, we cannot mount it as `com.tinyspeck.chatlyio@documents/appium.png`
+    - The others works as _format 2_
+        - Can be mounted apps which have `UIFileSharingEnabled` flag in its `info.plist`
+- `path_to_the_file_or_folder_inside_container` is the target to push/pull to/from them.
+    - If the `optional_container_type` is `documents`, this path will be mapped to
+      `On My iPhone/<app name>` in Files app
+
+_format 3_ is not allowed for real devices.
+
+#### Example
+
+If you would like to pull _Presentation.key_ form Keynote app, you can get it as below.
+
+- Pull file
+
+```javascript
+// webdriver.io
+let data = driver.pullFile('@io.appium.example:documents/Presentation.key');
+```
+
+```ruby
+# ruby_lib_core
+file = @driver.pull_file '@com.apple.Keynote:documents/Presentation.key'
+File.open('presentation.key', 'wb') { |f| f<< file }
```

Comment:
should the file be decoded from base64 first?

Reply:
No, the Ruby API returns base64 decoded data as same as push method. `push` also encodes inside the method because of https://github.com/appium/appium/pull/12710/files#r289629414